src/utils/partition.js

```diff
@@ -1,14 +1,14 @@
 "use strict";
 
+const reduce = Array.prototype.reduce;
+
 module.exports = function partition(array, fn) {
-  const a = [];
-  const b = [];
-  array.forEach(item => {
-    if (fn(item)) {
-      a.push(item);
-    } else {
-      b.push(item);
-    }
-  });
-  return [a, b];
+  return reduce.call(
+    array,
+    (result, item) => {
+      result[fn(item) ? 0 : 1].push(item);
```

Comment:
I personally find this harder to read.

Reply:
reduce is much reasonable,  and `lodash` has the exactly same implementation

https://github.com/lodash/lodash/blob/d5ef31929a1262abbc75b8dadc0b6ae6e9558b5f/partition.js#L26